pkg/apis/pingcap/v1alpha1/types.go

```diff
@@ -90,19 +91,64 @@ type TidbClusterList struct {
 // +k8s:openapi-gen=true
 // TidbClusterSpec describes the attributes that a user creates on a tidb cluster
 type TidbClusterSpec struct {
-	SchedulerName string   `json:"schedulerName,omitempty"`
-	PD            PDSpec   `json:"pd,omitempty"`
-	TiDB          TiDBSpec `json:"tidb,omitempty"`
-	TiKV          TiKVSpec `json:"tikv,omitempty"`
-	// +k8s:openapi-gen=false
-	TiKVPromGateway TiKVPromGatewaySpec `json:"tikvPromGateway,omitempty"`
```

Comment:
This field is not used anymore, removing this is backward compatible as long as the validation code do not reject this field